state/graphql-sender/src/main/java/com/expediagroup/streamplatform/streamregistry/state/graphql/DefaultApolloClientFactory.java

```diff
@@ -29,24 +29,50 @@
 public class DefaultApolloClientFactory implements ApolloClientFactory {
   @NonNull private final String streamRegistryUrl;
   @NonNull private final Consumer<ApolloClient.Builder> configurer;
+  private Credentials credentials;
 
   public DefaultApolloClientFactory(String streamRegistryUrl) {
     this(streamRegistryUrl, builder -> {});
   }
 
+  public DefaultApolloClientFactory(String streamRegistryUrl, Credentials credentials) {
+    this(streamRegistryUrl, builder -> {});
+    this.credentials = credentials;
+  }
+
+  public DefaultApolloClientFactory(String streamRegistryUrl, Consumer<ApolloClient.Builder> configurer, Credentials credentials) {
+    this(streamRegistryUrl, configurer);
+    this.credentials = credentials;
+  }
+
   @Override
   public ApolloClient create() {
-    var builder = builder()
-        .okHttpClient(new OkHttpClient.Builder().build());
+    OkHttpClient okHttpClient;
+
+    if (credentials != null) {
+      okHttpClient = new OkHttpClient.Builder()
+          .authenticator((route, response) -> response.request().newBuilder()
```

Comment:
We don't need the authenticator AND the interceptor - just the interceptor. The code example in the ticket was just illustrating both ways of doing it.

Reply:
done